code/optimize/src/main/java/com/adobe/marketing/mobile/optimize/OptimizeExtension.java

```diff
@@ -232,10 +233,78 @@ void handleOptimizeRequestContent(@NonNull final Event event) {
+                                        + " request event, provided list of decision scopes has no"
+                                        + " valid scope.");
+                        getApi().dispatch(
+                                        createResponseEventWithError(
+                                                event, AdobeError.UNEXPECTED_ERROR));
+                        return;
+                    }
+
+                    // Fetch propositions for the decision scopes from the cache
+                    Map<DecisionScope, OptimizeProposition> fetchedPropositions = new HashMap<>();
+                    for (DecisionScope scope : eventDecisionScopes) {
+                        if (cachedPropositions.containsKey(scope)) {
+                            fetchedPropositions.put(scope, cachedPropositions.get(scope));
+                        }
+                    }
+
+                    // Check if all scopes are cached and none are in progress
+                    boolean anyScopeInProgress = false;
+                    HashSet<DecisionScope> scopesInProgress = new HashSet<>();
+                    for (List<DecisionScope> updatingScope :
+                            updateRequestEventIdsInProgress.values()) {
+                        scopesInProgress.addAll(updatingScope);
+                    }
+                    for (DecisionScope scope : eventDecisionScopes) {
+                        if (scopesInProgress.contains(scope)) {
+                            anyScopeInProgress = true;
+                            break;
+                        }
+                    }
+
+                    if ((fetchedPropositions.size() == eventDecisionScopes.size())
+                            && !anyScopeInProgress) {
+                        Log.trace(
+                                OptimizeConstants.LOG_TAG,
+                                SELF_TAG,
+                                "handleOptimizeRequestContent - All scopes are cached and none are"
+                                        + " in progress, dispatching event directly.");
+
+                        // Dispatch the event directly
+                        handleGetPropositions(event);
+                    } else {
+                        Log.trace(
+                                OptimizeConstants.LOG_TAG,
+                                SELF_TAG,
+                                "handleOptimizeRequestContent - Scopes are not fully cached or are"
+                                        + " in progress, adding event to dispatcher.");
+                        eventsDispatcher.offer(event);
+                    }
+                    break;
+                } catch (final Exception e) {
+                    Log.warning(
+                            OptimizeConstants.LOG_TAG,
+                            SELF_TAG,
+                            "handleUpdatePropositions - Failed to process update propositions"
```

Comment:
nit: this needs to be `"handleOptimizeRequestContent - Failed to process get propositions"`